react/tic-tac-toe/src/redux/login/reducer.js

```diff
@@ -0,0 +1,31 @@
+import Immutable from 'seamless-immutable';
+
+import { actions } from './actions';
+
+const initialState = {
+  values: {},
+  token: ''
+};
+
+function reducer(state = Immutable(initialState), action) {
+  switch (action.type) {
+    case actions.GET_LOGIN:
+      return state.merge({
+        values: !state.values
+      });
+    case actions.GET_LOGIN_SUCCESS:
+      return state.merge({
+        values: action.payload.values,
+        token: action.payload.token,
+        error: null
+      });
```

Comment:
you don't have to put `error: null`here because the error here doesn`t have a value that you need to change.